### PR TITLE
feat(providers): add Whop

### PR DESCRIPTION
**Pull request by JedPattersonn**
> _NOTE_:
>
> - It's a good idea to open an issue first to discuss potential changes.
> - Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

## ☕️ Reasoning

Whop is a digital marketplace that allows developers to sell access to their software. This is done through OAuth.

Currently, developers are implementing this manually or using custom NextAuth configs. Adding Whop as an official provider will allow developers to easily sell access to their websites.

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: INSERT_ISSUE_LINK_HERE

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/core/src/providers/whop.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    profile(profile) {
+      return {
+        id: profile.id,
+        name: profile.username,
+        email: profile.email,
+        image: profile.profile_pic_url,
+      }
+    },
+    style: {
+      logo: "/google.svg",
```
Comment (ThangHuuVu) [inline packages/core/src/providers/whop.ts:90]:
```suggestion
      logo: "/whop.svg",
```

**Review by ThangHuuVu — COMMENTED** [review]
Thanks! Just one comment about copy pasta

**Comment by ThangHuuVu** [comment]
Oh sorry, I see that this is a WIP PR. Do ping me when you think it's ready to be reviewed again.

**Comment by ndom91** [comment]
We've got another PR open for adding Whop, can you take a look and maybe merge your efforts?

https://github.com/nextauthjs/next-auth/pull/8031

**Pull request closed without merging**